### PR TITLE
Update deprecated Azure things

### PR DESCRIPTION
**Pull request by rosslovas**
Certain Azure resources in use have been deprecated. This PR updates the terraform config to use the resources that supercede those deprecated resources.

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -47,22 +47,19 @@ resource "azurerm_resource_group" "group" {
...
 
-resource "azurerm_app_service" "web" {
-  name                = "github-status-checks-${lower(var.environment)}"
-  location            = azurerm_resource_group.group.location
+resource "azurerm_windows_web_app" "web" {
+  name                = "github-status-checks-${lower(var.environment)}-core-platform"
+  location            = azurerm_service_plan.plan.location
   resource_group_name = azurerm_resource_group.group.name
-  app_service_plan_id = azurerm_app_service_plan.plan.id
+  app_service_plan_id = azurerm_service_plan.plan.id
```
Comment (akirayamamoto) [inline terraform/main.tf:62]: I think you need to change this to service_plan_id

Comment (akirayamamoto) [reply terraform/main.tf:62]: See the example here https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/windows_web_app

**Review by akirayamamoto — APPROVED** [review]
There's one thing to fix, otherwise looks good, thank you.

File: terraform/main.tf (hunk 1 above)
Comment (rosslovas) [reply terraform/main.tf:62]: You're right, changing now, thank you!

**Pull request merged**